### DISCUSSION TITLE
V9: Drawer Component as SidePanel

### DISCUSSION THREAD

**Discussion by blendsdk** [microsoft/fluentui#29512] [Q&A] [answered]
Hi,

First of all thank you for providing the Drawer component. I was wondering If there is a way to customize the Drawer to mimic a side-panel component when closed for it to show Icons? The Drawer is capable of being Inline, but would it possible for the Drawer to not entirely close? This way perhaps I can render a vertical toolbar in the Drawer.

I have a similar component for v8 but it would be nice to have a built-in component in v9 instead.
The following is my current component.

![Screenshot 2023-10-12 at 15 11 13](https://github.com/microsoft/fluentui/assets/5524595/5a13697d-e02b-4e91-b7fb-687e0ac0c9d1)

![Screenshot 2023-10-12 at 15 11 21](https://github.com/microsoft/fluentui/assets/5524595/21cbc1ed-bbc2-474c-a794-414509a51562)

**Comment by blendsdk** [comment]
**Update:** 

I have created a custom `SidePanelLayout` with a custom `Drawer` in v9 for now. Hopefully we will get something permanent in the future. It was surprisingly easy when incorporating the `tokens` for CSS and layout. 

![Screenshot 2023-10-15 at 20 50 25](https://github.com/microsoft/fluentui/assets/5524595/1169f37c-4f44-48c6-8e3d-c51e06f40323)

**Comment by frankyphs** [comment]
Sir, may i ask the code for my privacy learning Sir? Iam learning fluent UI V9 now. Thanks

**Comment by marcosmoura** [comment, marked as answer]
Currently we don't have any plans to support a "mini" Drawer variant. We will discuss internally the implications and usages of such implementation, but we cannot promise any ETA. :)
But that can be easily implemented using an "always open" inline Drawer. Here is an example:
https://codesandbox.io/s/objective-forest-vx2j96?file=/example.tsx
